python/pyproject.toml

```
@@ -15,20 +15,7 @@ classifiers = [
   "Programming Language :: Python :: 3",
 ]
 
-dependencies = [
-  "requests==2.30.0",
-  "black",
-  "flake8",
-  "pytest",
-  "pydantic>=2.1",
-  "pybars3",
-  "google-generativeai",
-  "openai>=1.0.0",
-  "python-dotenv",
-  "huggingface_hub",
-  "result",
-  "nest_asyncio",
-  ]
+dependencies = {file = ["requirements.txt"]}
```

```suggestion
dynamic = ["dependencies"]

[tool.setuptools.dynamic]
dependencies = {file = ["requirements.txt"]}
```

Something faced in Semantic Retrieval earlier - when pip installing, needed to specify the dynamic stuff to read from requirements.txt properly - https://github.com/lastmile-ai/semantic-retrieval/blob/main/python/pyproject.toml#L17